src/Form/LegacyConsumer/Commands/Helper/ClosureProxy.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Give\Form\LegacyConsumer\Commands\Helper;
+
+class ClosureProxy {
+
+	/**
+	 * @param callable $callback
+	 * @param HookCommandInterface $command
+	 */
+	public function __construct( callable $callback, $command ) {
+		$this->callback = $callback;
+		$this->command  = $command;
+	}
+
+	public function with( $attributes = [] ) {
+
+		// Merge command closure properties with attributes.
+		$reflection = new \ReflectionClass( $this->command );
+		foreach ( $reflection->getProperties() as $property ) {
+			if ( is_a( $this->command->{$property->name}, ClosureProxy::class ) ) {
+				$attributes[ $property->name ] = $this->command->{$property->name};
+			}
+		}
```

Comment:
Not sure I'm quite following the purpose in this.

Reply:
This makes all callbacks of the class available to each callback so that nested callbacks still have access to the class properties.